compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1168,7 +1166,7 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                     })
                 }
                 ty::Param(param) => {
-                    let generics = self.tcx.generics_of(hir_id.owner.to_def_id());
+                    let generics = self.tcx.generics_of(body_id);
```

Comment:
This is not exactly equivalent for closures, but the difference doesn't matter.
r=me once CI passes